bootconfig/README.md

```diff
@@ -0,0 +1,53 @@
+# Bootconfig API
+
+The bootconfig service is responsible for the setting of bootstrap configuration for the device after initial Bootz bootstrap.
+This message may include both the base device configuration but also the new gNSI artifacts.
+
+While this is service which can be used out of band it also allows for the development of an industry standard CLI for recovery
+of devices.
+
+## RPC use case
+
+* From remote rpc client call `gnoi.Bootconfig.Set` using bootconfig message containing elements to override.
+
+## CLI use cases
+
+The CLI should be bundled with the router and exist in the default path.
+The CLI must be named `oc_cli` it must have the following heirarchy for calling specific services.
+
+```
+oc_cli
+  gnoi
+    bootconfig
+    factoryreset
+  gnmi
+    get
+    set
+    subscribe
+```
```

Comment:
Are these real gnmi/gnoi calls, e.g. there's a gnmi and gnoi clients on the node that can be accessed using this brand new cli? And, correspondingly, there's a grpc server that listens for these commands?

Reply:
no they would use the same expected library that the server on device would use for consistency but would not be an RPC call to those services as the point of this cli is to bypass those services not working due to auth or other system configuration issues